src/github.com/getlantern/flashlight/statreporter/statreporter.go

```diff
@@ -147,11 +147,23 @@ func (r *reporter) run() {
 					categoryStats = make(stats)
 					dgAccum.categories[update.category] = categoryStats
 				}
-				switch update.action {
+				switch a := update.action.(type) {
```

Comment:
I had to do some googling to figure out exactly what was going on here, which led me to http://blog.denevell.org/golang-interface-type-assertions-switch.html -- I didn't even know about type switches! This is super cool and useful when you're working with an interface{} type.